web/src/utils.js

```diff
@@ -41,60 +41,85 @@ const partition = (collection, filter) => {
 };
 
 /**
- * Allows using the React useEffect hook in a safer way.
+ * @typedef {Object} cancellableWrapper
+ * @property {Promise} promise - Cancellable promise
+ * @property {function} cancel - Function for canceling the promise
+ */
+
+/**
+ * Creates a wrapper object with a cancellable promise and a function for canceling the promise
+ *
+ * @see useCancellablePromise
+ *
+ * @param {Promise} promise
+ * @returns {cancellableWrapper}
+ */
+function makeCancellable(promise) {
+  let isCanceled = false;
+
+  const cancellablePromise = new Promise((resolve, reject) => {
+    promise
+      .then((value) => (!isCanceled && resolve(value)))
+      .catch((error) => (!isCanceled && reject(error)));
+  });
+
+  return {
+    promise: cancellablePromise,
+    cancel() {
+      isCanceled = true;
+    }
+  };
+}
+
+/**
+ * Allows using promises in a safer way.
  *
- * This effect is useful for performing actions that modify a React component after resolving a
- * promise (e.g., setting the component state once a D-Bus call is answered). Note that nothing
+ * This hook is useful for safely performing actions that modify a React component after resolving
+ * a promise (e.g., setting the component state once a D-Bus call is answered). Note that nothing
  * guarantees that a React component is still mounted when a promise is resolved.
  *
  *  @see {@link https://overreacted.io/a-complete-guide-to-useeffect/#speaking-of-race-conditions|Race conditions}
  *
- * This effect receives a callback function as argument. That callback will be invoked passing a
- * function argument which can be used inside the callback code in order to run unsafe actions in a
- * safe way.
- *
- * The callback is the only one dependency of this effect. Make sure the callback object does not
- * change to avoid firing the effect after every completed render. It is recommended to wrap your
- * callback function with a useCallback hook.
+ * The hook provides a function for making promises cancellable. All cancellable promises are
```

Comment:
np: I would clarify that the promise is not *really* canceled and we are just not calling resolve/reject.